code/app/src/main/java/com/example/the_tarlords/MainActivity.java

```diff
@@ -40,6 +59,10 @@ public class MainActivity extends AppCompatActivity {
     private static String userId;
     private static View hView;
     private Object lock = new Object();
+    private FusedLocationProviderClient client;
+    private static final int REQUEST_LOCATION_PERMISSION = 99;
+    // Need this for location
+    private String eventId;
```

Comment:
this isn't really an attribute of MainActivity as a whole, would probably be better to define it as an attribute of individual methods if possible, if this is the only way that works though then just leave it